internal/otlpjson/marshaler.go

```diff
@@ -0,0 +1,69 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package otlpjson
+
+import (
+	"bytes"
+
+	"github.com/gogo/protobuf/jsonpb"
+
+	otlpcollectorlogs "go.opentelemetry.io/collector/internal/data/protogen/collector/logs/v1"
+	otlpcollectormetrics "go.opentelemetry.io/collector/internal/data/protogen/collector/metrics/v1"
+	otlpcollectortrace "go.opentelemetry.io/collector/internal/data/protogen/collector/trace/v1"
+	"go.opentelemetry.io/collector/internal/model/serializer"
+)
+
+type marshaler struct {
+	delegate jsonpb.Marshaler
+}
+
+// NewTracesMarshaler returns a serializer.TracesUnmarshaler to marshal to OTLP json bytes.
+func NewTracesMarshaler() serializer.TracesMarshaler {
+	return &marshaler{delegate: jsonpb.Marshaler{}}
+}
+
+// NewMetricsMarshaler returns a serializer.MetricsMarshaler to marshal to OTLP json bytes.
+func NewMetricsMarshaler() serializer.MetricsMarshaler {
+	return &marshaler{delegate: jsonpb.Marshaler{}}
+}
+
+// NewLogsMarshaler returns a serializer.LogsMarshaler to marshal to OTLP json bytes.
+func NewLogsMarshaler() serializer.LogsMarshaler {
+	return &marshaler{delegate: jsonpb.Marshaler{}}
+}
+
+func (en *marshaler) MarshalLogs(model interface{}) ([]byte, error) {
+	buf := bytes.Buffer{}
+	if err := en.delegate.Marshal(&buf, model.(*otlpcollectorlogs.ExportLogsServiceRequest)); err != nil {
+		return nil, err
+	}
+	return buf.Bytes(), nil
+}
+
+func (en *marshaler) MarshalMetrics(model interface{}) ([]byte, error) {
+	buf := bytes.Buffer{}
+	if err := en.delegate.Marshal(&buf, model.(*otlpcollectormetrics.ExportMetricsServiceRequest)); err != nil {
```

Comment:
Do we want these to be panics if the wrong type is passed in? See here for how it checks the type:

https://github.com/open-telemetry/opentelemetry-collector/pull/3242/files#diff-99f0341981a504978dfba25f8c54cbaff15a4ef2022f7e4b72ee35fd25835787R46

It should _probably_ be caught with tests if wrong type is passed but I'm not 100% sure how these API's will be used which is why I went the safer route.

Reply:
Though I was kind of thinking for OTLP the model would still be pdata. (ie the translation step would be a nop, passing through pdata) which the marshaler would then unwrap to the protobuf before serializing.

Reply:
nm, this probably makes more sense.

Reply:
Done.